src/AsyncBridge/AsyncCompatLibExtensions.cs

```diff
@@ -291,7 +294,8 @@ private static bool TryRemoveTimer(this ConcurrentDictionary<WeakCTS, WeakRefere
 
         timer = (Timer)weakReference.Target;
 
-        return timer != null; // Should always return true, but to be safe we check
+        // Will always return true, since this is only called from the Timer callback, which will have a strong reference to the Timer and the CTS
+        return timer != null;
```

Comment:
Sorry I didn't say this earlier... why not `return true`? 😇

Reply:
`Should always return true`, then. But that's quite a nitpick and I don't really care. 😁